src/main/java/com/minecolonies/coremod/client/gui/WindowGuardControl.java

```diff
@@ -194,14 +201,17 @@ public void onUpdate()
     {
         super.onUpdate();
 
-        pullInfoFromHut();
-
-        if (!task.equals(GuardTask.PATROL))
+        if (rand.nextInt(20) < 1)
         {
-            listOfPoints.hide();
+            pullInfoFromHut();
+
+            if (!task.equals(GuardTask.PATROL))
+            {
+                listOfPoints.hide();
+            }
+            window.findPaneOfTypeByID(GUI_ELEMENT_LIST_LEVELS, ScrollingList.class).refreshElementPanes();
+            workersListPane.refreshElementPanes();
```

Comment:
this does not stop the element on update so you basically are triggered that twice the same tick

Reply:
this is the only lag source in the log though